sumpy/expansion/__init__.py

```diff
@@ -282,12 +282,11 @@ def _split_coeffs_into_hyperplanes(self) -> List[Tuple[int, List[Tuple[int]]]]:
                 pass
             else:
                 # Calculate the multi-index that appears last in in the PDE in
-                # reverse degree lexicographic order (degrevlex).
-                # The monomial ordering used here which is degrevlex, must match
-                # the monomial ordering used in LinearPDEBasedExpansionTermsWrangler
-                max_mi_idx = max(mi_to_index[ident.mi] for
-                                 ident in pde_dict.keys())
-                max_mi = mis[max_mi_idx]
+                # degree lexicographic order.
+                ordering_key = self._get_mi_ordering_key()
```

Comment:
It looks like `_get_mi_ordering_key` is only defined in `LinearPDEBasedExpansionTermsWrangler`. Wouldn't this make `ExpansionTermsWrangler._split_coeffs_into_hyperplanes` unusable unless we're in that subclass? Maybe this wants to be a function instead?

Reply:
Fixed in 566afb7 by adding a method that the subclass overrides.